packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -212,8 +212,16 @@ class TextPainter {
     canvas.drawParagraph(_paragraph, offset);
   }
 
+  bool _isUtf16Surrogate(int value) {
+    return value >= 0xd800 && value <= 0xdfff;
```

Comment:
I think `(value & 0xD800 == 0xD800)` is equivalent, and may be microscopically faster?